From stable-bounces@linux.kernel.org Tue Oct 17 04:50:35 2006
Date: Mon, 16 Oct 2006 15:59:28 GMT
Message-Id: <200610161559.k9GFxSnJ005249@hera.kernel.org>
From: Marcel Holtmann <marcel@holtmann.org>
MIME-Version: 1.0
Subject: Bluetooth: Check if DLC is still attached to the TTY
Content-Type: text/plain; charset="us-ascii"

From: Marcel Holtmann <marcel@holtmann.org>

[Bluetooth] Check if DLC is still attached to the TTY

If the DLC device is no longer attached to the TTY device, then it
makes no sense to go through with changing the termios settings.

Signed-off-by: Marcel Holtmann <marcel@holtmann.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 net/bluetooth/rfcomm/tty.c |    3 +++
 1 file changed, 3 insertions(+)

--- linux-2.6.18.1.orig/net/bluetooth/rfcomm/tty.c
+++ linux-2.6.18.1/net/bluetooth/rfcomm/tty.c
@@ -748,6 +748,9 @@ static void rfcomm_tty_set_termios(struc
 
 	BT_DBG("tty %p termios %p", tty, old);
 
+	if (!dev)
+		return;
+
 	/* Handle turning off CRTSCTS */
 	if ((old->c_cflag & CRTSCTS) && !(new->c_cflag & CRTSCTS)) 
 		BT_DBG("Turning off CRTSCTS unsupported");
